tarfs_test.go

```diff
@@ -0,0 +1,87 @@
+package tarfs
+
+import (
+	"bytes"
+	"encoding/hex"
+	"io"
+	"io/ioutil"
+	"testing"
+
+	"github.com/getlantern/testify/assert"
+)
+
+func TestRoundTrip(t *testing.T) {
+	expectedA, err := ioutil.ReadFile("resources/a.txt")
+	if err != nil {
+		t.Fatalf("Unable to load expectedA: %v", err)
+	}
+
+	expectedB, err := ioutil.ReadFile("localresources/b.txt")
+	if err != nil {
+		t.Fatalf("Unable to load expectedB: %v", err)
+	}
+
+	tarString := bytes.NewBuffer(nil)
+	err = EncodeToTarString("resources", tarString)
+	if err != nil {
+		t.Fatalf("Unable to encode to tar string: %v", err)
+	}
+
+	fs, err := New(tarStringToBytes(t, tarString), "localresources")
+	if err != nil {
+		t.Fatalf("Unable to open filesystem: %v", err)
+	}
+
+	a, err := fs.Get("a.txt")
+	if assert.NoError(t, err, "a.txt should have loaded") {
+		assert.Equal(t, string(expectedA), string(a), "A should have matched expected")
+	}
+
+	b, err := fs.Get("b.txt")
+	if assert.NoError(t, err, "b.txt should have loaded") {
+		assert.Equal(t, string(expectedB), string(b), "B should have matched expected")
+	}
+
+	f, err := fs.Open("/nonexistentdirectory/")
+	if assert.NoError(t, err, "Opening nonexistent directory should work") {
+		fi, err := f.Stat()
+		if assert.NoError(t, err, "Should be able to stat directory") {
+			assert.True(t, fi.IsDir(), "Nonexistent directory should be a directory")
+		}
+	}
+
+	f, err = fs.Open("/nonexistentfile")
+	assert.Error(t, err, "Opening nonexistent file should fail")
+
+	f, err = fs.Open("/a.txt")
+	if assert.NoError(t, err, "Opening existing file should work") {
+		fi, err := f.Stat()
+		if assert.NoError(t, err, "Should be able to stat file") {
+			if assert.False(t, fi.IsDir(), "File should not be a directory") {
+				if assert.Equal(t, len(expectedA), fi.Size(), "File info should report correct size") {
+					a := bytes.NewBuffer(nil)
+					_, err := io.Copy(a, f)
+					if assert.NoError(t, err, "Should be able to read from file") {
+						assert.Equal(t, expectedA, a.Bytes(), "Should have read correct data")
+					}
+				}
+			}
+		}
+	}
+}
+
```

Comment:
Though not support directory, we do and should support files under subdirectory, add test to cover it?